src/BudgetManager/BudgetManager.API/Controllers/V1/TransactionController.cs

```diff
@@ -15,7 +15,23 @@ namespace BudgetManager.API.Controllers.V1
     public class TransactionController : ControllerBase
     {
         private readonly IMediator _mediator;
-        public TransactionController(IMediator mediator) => _mediator = mediator;
+        private readonly AddExpenseTransactionValidator _addExpenseValidator;
+        private readonly UpdateExpenseTransactionValidator _updateExpenseValidator;
+        private readonly AddIncomeTransactionValidator _addIncomeValidator;
+        private readonly UpdateIncomeTransactionValidator _updateIncomeValidator;
+
+        public TransactionController(IMediator mediator,
+            AddExpenseTransactionValidator addExpenseValidator,
+            UpdateExpenseTransactionValidator updateExpenseValidator,
+            AddIncomeTransactionValidator addIncomeValidator,
+            UpdateIncomeTransactionValidator updateIncomeValidator)
```

Comment:
I think it would be better to have something like ITransactionValidator and injecting it with DI.

Reply:
these classes inherit from different DTO . and also they use Base repository<User> in ctor. when I tried inject with interfaces they did not receive Base repo and thew Exception. that's why I asked if I can combine add and update DTOs into one common DTO and have less Validator classes

Reply:
If you meant loose dependency. they are Scoped lifeSpan by default.